pkg/storage/config.go

```diff
@@ -0,0 +1,40 @@
+package storage
+
+import (
+	"context"
+)
+
+// PostgresConfig contains the Postgres database configuration arguments
+type PostgresConfig struct {
+	Hostname     string
+	Port         string
+	Username     string
+	Password     string
+	DatabaseName string
+}
+
+// Name is used by the settings library to replace the default naming convention.
+func (c *PostgresConfig) Name() string {
+	return "Postgres"
+}
+
+// PostgresConfigComponent satisfies the settings library Component API,
+// and may be used by the settings.NewComponent function.
+type PostgresConfigComponent struct{}
+
+// Settings populates a set of defaults if none are provided via config.
+func (*PostgresConfigComponent) Settings() *PostgresConfig {
+	return &PostgresConfig{}
+}
+
+// New constructs a PostgresConfig from a config.
+func (*PostgresConfigComponent) New(_ context.Context, c *PostgresConfig) (*PostgresConfig, error) {
```

Comment:
Based on the [docs](https://github.com/asecurityteam/settings) I still don't think this is the correct usage of the settings Factory. Given: 

```
// All configuration is driven by a struct that uses standard go types.
type Config struct {}
// The resulting element is virtually anything that you need it to be
// for the purposes of the system you are building.
type Result struct {}
// The component contract interfaces between the thing you want to make,
// the result, and the settings project. It is responsible for producing
// instances of the configuration struct with any default values populated.
// It is also used to construct new instances of the result using a
// populated configuration. No references to the settings project are
// required to create any part of this setup.
type Component struct {}
func (*Component) Settings() *Config { return &Config{} }
func (*Component) New(_ context.Context, c *Config) (*Result, error) {
    return &Result{}, nil
}
```

Our `DB` struct should take the place of the `Result` struct. @kconwayatlassian can you provide some guidance on the usage here?

Reply:
I'll copy my previous comment since it was lost in the change:
`settings` components are effectively factories, or factory factories, that get interfaced with a special configuration loader. The `Settings()` method should return the configuration struct with any default values set and the `New` method should return the actual thing being constructed. In this case, I suggested we use a component because it enables us to make the concrete storage implementation selectable and configurable via configuration. The output should be an implementation of whatever interface represents the storage backend.

Reply:
That's fine.  I'll change my impl to reflect the intent of the Settings/Component lib.